Updating "Full metrics pipeline" para(resource-usage-monitoring.md Modified) (#38816)
kubernetes/website: content/en/docs/tasks/debug/debug-cluster/resource-usage-monitoring.md

then exposes them to Kubernetes via an adapter by implementing either the
`custom.metrics.k8s.io` or `external.metrics.k8s.io` API.

Kubernetes is designed to work with [OpenMetrics](https://openmetrics.io/), 
which is one of the
[CNCF Observability and Analysis - Monitoring Projects](https://landscape.cncf.io/card-mode?category=monitoring&project=graduated,incubating,member,no&grouping=category&sort=stars),
built upon and carefully extending [Prometheus exposition format](https://prometheus.io/docs/instrumenting/exposition_formats/)
in almost 100% backwards-compatible ways.

> If you glance over at 

Reviewer: (nit)
```suggestion
If you glance over at the
```

then exposes them to Kubernetes via an adapter by implementing either the
`custom.metrics.k8s.io` or `external.metrics.k8s.io` API. 

> Kubernetes is designed to work with [OpenMetrics](https://openmetrics.io/), 

Reviewer 2: ```suggestion
Kubernetes is designed to work with [OpenMetrics](https://openmetrics.io/), 
```
I don't think this statement is correct
/hold

which is one of the
[CNCF Observability and Analysis - Monitoring Projects](https://landscape.cncf.io/card-mode?category=monitoring&project=graduated,incubating,member,no&grouping=category&sort=stars),
built upon and carefully extending [Prometheus exposition format](https://prometheus.io/docs/instrumenting/exposition_formats/)
in almost 100% backwards-compatible ways.

> If we glance over at 

Reviewer 3: ```suggestion
If you glance over at 
```

[CNCF Landscape](https://landscape.cncf.io/card-mode?category=monitoring&project=graduated,incubating,member,no&grouping=category&sort=stars)

> we can observe that Kubernetes project doesn't provide a recommendation 

Reviewer 3: ```suggestion
you can observe that Kubernetes project doesn't provide a recommendation 
```

Author: I think this one is correct, isn't it @NitishKumar06 . it says if you glance over at CNCF Landscape you can observe that Kubernetes------- if I add It is important to mention it won't make sense with the previous sentence.
Do correct me if I'm wrong.
Regards

Reviewer 3: According to style guides, we prefer using _you_ rather than _we_.

Author: I have changed 'we' to 'you' wherever it was used and yes it does look more readable now. Thanks for the suggestion 😁

Reviewer 3: I'm not sure what text in the docs helps in **_observing_** that Kubernetes doesn't provide a recommendation......
I'd still go with the text _It is important to mention that......_

Author: So how to add the CNCF landscape line with It is imp to mention any suggestions? Maybe we can write this Glancing over at CNCF Landscape, It is imp to mention----- is this good??

Author: While you glance over at CNCF Landscape, It is important to mention that the Kubernetes project doesn't provide recommendations.
I think this works

Reviewer 3: sounds okayish! Though I'd not provide a lgtm tag until this PR goes through another set of review by other reviewers & maintainers.

> you can observe that the Kubernetes project doesn't provide a recommendation 

Reviewer 3: ```suggestion
you can observe that the Kubernetes project doesn't provide a recommendation 
```
just a quick question or more like a doubt, how can you say that k8s don't provide a recommendation of a set of projects **just by looking at the landscape**? There is no set of lines that gives this kind of proof. Maybe we can modify this!

Author: Maybe because when we look at CNCF landscape there are tons of tools to work with k8s, not only Prometheus. One can check out other tools and work with one they think suits them best.

Author: ![image](https://user-images.githubusercontent.com/101946115/216019003-a16fd1be-086a-4194-ba14-5c9b766513c1.png)

Like if I'm looking over at Monitoring, I can see there are many projects other than Prometheus, and one can check out these and work with the one they think suits them best like Open Metrics or Thanos for example

> of projects to work upon as it is totally up to the users to find the 

Reviewer 3: ```suggestion
of any projects to work on, as it is totally up to the users to find the 
```

> project that suites them best. If someone is to make a full metrics pipeline,

Reviewer 3: ```suggestion
project that suites them the best. If someone is to make a full metrics pipeline,
```

> project that suites them the best. If someone is to make a full metrics pipeline,

Reviewer 3: ```suggestion
project that suites them the best. If someone is to make a full metrics pipeline,
```
How about providing a reference link to _full metrics pipeline_?

> you can observe that the Kubernetes project doesn't provide a recommendation 
> of any projects to work on, as it is totally up to the users to find the 
> project that suits them the best. If someone is to make a full metrics pipeline,
> they need to design and deploy their own infrastructure platform, of which
> Kubernetes is only one part.

Reviewer: I'd write:
```suggestion
you can see a number of monitoring projects that can work with Kubernetes by _scraping_
metric data and using that to help you observe your cluster. It is up to you to select the tool
or tools that suit your needs. The CNCF landscape for observability and analytics includes a
mix of open-source software, paid-for software-as-a-service, and other commercial products.

When you design and implement a full metrics pipeline you can make that monitoring data
available back to Kubernetes. For example, a HorizontalPodAutoscaler can use the processed
metrics to work out how many Pods to run for a component of your workload.
```

Author: Ok, this sounds way better than mine 😂. @NitishKumar06 what's your opinion on this??

Reviewer: I would go with that suggestion @ShivangShandilya (I'm writing as a SIG Docs reviewer).

Reviewer 3: Proceed with the suggestions given by @sftim

then exposes them to Kubernetes via an adapter by implementing either the
`custom.metrics.k8s.io` or `external.metrics.k8s.io` API. 

> Kubernetes is designed to work with [OpenMetrics](https://openmetrics.io/), which is one of the [CNCF Observability and Analysis - Monitoring Projects](https://landscape.cncf.io/card-mode?category=monitoring&project=graduated,incubating,member,no&grouping=category&sort=stars), built upon and carefully extending  [Prometheus exposition format](https://prometheus.io/docs/instrumenting/exposition_formats/) in almost 100% backwards-compatible ways.. 

Reviewer 4: two dots at the end?

Author: Resolved

> If we glance over at [CNCF Landscape](https://landscape.cncf.io/card-mode?category=monitoring&project=graduated,incubating,member,no&grouping=category&sort=stars) we can observe that Kubernetes project doesn't provide a recommendation of projects to work upon as it is totally upto the users to find the project that suites them best. If someone is to make a full metrics pipeline, they need to design and deploy their own infrastructure platform, of which Kubernetes is only one part.

Reviewer 4: upto is not a valid word?

"glance over at ..."?

Author: upto was a typo error which I have corrected to - up to
and I think glace over at CNCF landscape is correct, do tell if I'm wrong.
Regards

> 

Reviewer 4: Please wrap the long lines manually when appropriate.
You can check how other paragraphs are formatted in markdown.

Author: Resolved
